azurelinuxagent/common/protocol/wire.py

```diff
@@ -162,7 +162,7 @@ def download_ext_handler_pkg(self, uri, destination, headers=None, use_proxy=Tru
         host_func = lambda: self._download_ext_handler_pkg_through_host(uri, destination)
 
         try:
-            success = self.client.send_request_using_appropriate_channel(direct_func, host_func)
+            success = self.client.send_request_using_appropriate_channel(direct_func, host_func) is not None
         except Exception:
             success = False
```

Comment:
Additionally looks like this wont every throw now, we can remove this except check maybe as its redundant?

Reply:
I had initially changed this and the caller of this method (since its try/catch is also redundant), but Norberto and I agreedto keep this as-is now so we contain the changes to the level of `send_request_using_appropriate_channel` and methods below. This way, the review is relatively simple without having to go up the call stack.

There's still a lot of improvements to be done with these internal functions (unifying their interfaces for one), so your comment can be addressed in a future round of changes.

Reply:
Umm I get the point of making less changes but in this particular instance the code path (line 166) will never be hit (because the underlying function can never raise after these changes). Given that, wouldn't it make sense to just remove that? Why bother with having more dead code to eventually deal with later when we can just do it right now :) 

>This way, the review is relatively simple without having to go up the call stack.

True this, I wish I had read this before but since I already went deep I figured I'd add the comment haha. 
I'll leave it upto you in that case :)